### ISSUE TITLE
Implement a Federated Learning Approach

### ISSUE THREAD

**Issue by rohaan2614** [rohaan2614/Federated_Learning#9]
Baseline's been implemented and it's working as well. Time to move to the distributed part.

**Comment by rohaan2614** [comment]
Accuracy is too low. Needs to be at least comparable to baseline.

**Comment by rohaan2614** [comment]
Haibo's Comment: use model.parameters() to get model weights, not model.state_dict()
